Source/CombatExtended/CombatExtended/ArmorUtilityCE.cs

```diff
@@ -347,6 +357,61 @@ private static DamageInfo GetDeflectDamageInfo(DamageInfo dinfo, BodyPartRecord
             return newDinfo;
         }
 
+        /// <summary>
+        /// Creates a new DamageInfo for blunt damage based off the blocked amount of damage and penetration of the supplied DamageInfo.
+        /// </summary>
+        /// <param name="dinfo">The damage info the additional blunt damage will be based off</param>
+        /// <param name="hitPart">The originally hit part</param>
+        /// <param name="dmgAmount">The amount of damage blocked</param>
+        /// <param name="penAmount">The amount of penetration blocked</param>
+        /// <returns>New DamageInfo based off dinfo with damage, penetration, Def and forceHitPart adjusted</returns>>
+        private static DamageInfo GetPartialPenetrationBluntDamageInfo(DamageInfo dinfo, BodyPartRecord hitPart, float dmgAmount, float penAmount) // Not referencing the float values directly since we don't want to update them

```

Comment:
This and GetDeflectDamageInfo have a lot of duplicate code between them, they should be merged into a single function that handles both use cases. I included an example of a modified GetDeflectDamageInfo function that incorporates your changes, but it's not tested at all, so if you integrate it wholesale, please make sure it works as expected:
https://gist.github.com/LockdownX7/45b299ba14c23b608a80b22b618f3a86

Reply:
Merged them both into one thanks to you, although I did modify it to work correctly.